src/ecalc/libraries/libecalc/common/libecalc/common/utils/rates.py

```diff
@@ -230,6 +232,27 @@ def __setitem__(
                     f"Could not update timeseries, Combination of indices of type '{type(indices)}' and values of type '{type(values)}' is not supported"
                 )
 
+    def __reindex_time_vector__(
```

Comment:
looks good. But why make it a dunder method? This usually indicates some kind of magic invocation using an operator etc.

We should also further refactor compressors to use this method instead of the static method and the bit cumbersome way that they handle reindexing. If there is a unit test for reindexing etc there, then reuse that for this, otherwise we must create unit tests for this reindex method. That can be done in a follow-up PR though with focus on making the compressor and pump reindexing (and others) work the same way).

Reply:
It should not be a dunder method, will fix:-)